dts/bindings/mtd/partition.yaml

```diff
@@ -16,6 +16,13 @@ properties:
 child-binding:
     description: Flash partition child node
     properties:
+       mount-type:
```

Comment:
should be `zephyr,` prefixed.  Would `zephyr,fs-type` make more sense?

Is there a reason this needs to be a string?  can we not have it be an `int` and have it match/utilize the FS_{FATFS, LITTLEFS}, etc enum?

Would really like to see it be an enum, but that raises question about external filesystems.

Reply:
Yes, it's a string specifically to make this useful to external file systems.

Reply:
Why can't this use the same int that is passed to `fs_register`?

Reply:
The issue has been addressed by this comment: https://github.com/zephyrproject-rtos/zephyr/pull/28320#discussion_r487142503